pkg/sdk/applications_impl_gen.go

```diff
@@ -0,0 +1,180 @@
+	if err != nil {
+		return nil, err
+	}
+	resultList := convertRows[applicationRow, Application](dbRows)
+	return resultList, nil
+}
+
+func (v *applications) ShowByID(ctx context.Context, id AccountObjectIdentifier) (*Application, error) {
+	request := NewShowApplicationRequest().WithLike(&Like{String(id.Name())})
+	applications, err := v.Show(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+	return collections.FindOne(applications, func(r Application) bool { return r.Name == id.Name() })
+}
+
+func (v *applications) Describe(ctx context.Context, id AccountObjectIdentifier) ([]ApplicationDetail, error) {
+	opts := &DescribeApplicationOptions{
+		name: id,
+	}
+	rows, err := validateAndQuery[applicationDetailRow](v.client, ctx, opts)
+	if err != nil {
+		return nil, err
+	}
+	return convertRows[applicationDetailRow, ApplicationDetail](rows), nil
+}
+
+func (r *CreateApplicationRequest) toOpts() *CreateApplicationOptions {
+	opts := &CreateApplicationOptions{
+		name:        r.name,
+		PackageName: r.PackageName,
+
+		DebugMode: r.DebugMode,
+		Comment:   r.Comment,
+		Tag:       r.Tag,
+	}
+	if r.Version != nil {
+		opts.Version = &ApplicationVersion{
+			VersionDirectory: r.Version.VersionDirectory,
+		}
+		if r.Version.VersionAndPatch != nil {
+			opts.Version.VersionAndPatch = &VersionAndPatch{
+				Version: r.Version.VersionAndPatch.Version,
+				Patch:   r.Version.VersionAndPatch.Patch,
+			}
+		}
+	}
+	return opts
+}
+
+func (r *DropApplicationRequest) toOpts() *DropApplicationOptions {
+	opts := &DropApplicationOptions{
+		IfExists: r.IfExists,
+		name:     r.name,
+		Cascade:  r.Cascade,
+	}
+	return opts
+}
+
+func (r *AlterApplicationRequest) toOpts() *AlterApplicationOptions {
+	opts := &AlterApplicationOptions{
+		IfExists: r.IfExists,
+		name:     r.name,
+
+		UnsetComment:                 r.UnsetComment,
+		UnsetShareEventsWithProvider: r.UnsetShareEventsWithProvider,
+		UnsetDebugMode:               r.UnsetDebugMode,
+		Upgrade:                      r.Upgrade,
+
+		SetTags:   r.SetTags,
+		UnsetTags: r.UnsetTags,
+	}
+	if r.Set != nil {
+		opts.Set = &ApplicationSet{
+			Comment:                 r.Set.Comment,
+			ShareEventsWithProvider: r.Set.ShareEventsWithProvider,
+			DebugMode:               r.Set.DebugMode,
+		}
+	}
+	if r.UpgradeVersion != nil {
+		opts.UpgradeVersion = &ApplicationVersion{
+			VersionDirectory: r.UpgradeVersion.VersionDirectory,
+		}
+		if r.UpgradeVersion.VersionAndPatch != nil {
+			opts.UpgradeVersion.VersionAndPatch = &VersionAndPatch{
+				Version: r.UpgradeVersion.VersionAndPatch.Version,
+				Patch:   r.UpgradeVersion.VersionAndPatch.Patch,
+			}
+		}
+	}
+	if r.UnsetReferences != nil {
+		opts.UnsetReferences = &ApplicationReferences{}
+		if r.UnsetReferences.References != nil {
+			s := make([]ApplicationReference, len(r.UnsetReferences.References))
+			for i, v := range r.UnsetReferences.References {
+				s[i] = ApplicationReference{
```

Comment:
+1 for reviewdog comment here

Reply:
i think this is fixed, but do we have a make target to run linting errors created by reviewdog locally?

Reply:
we have `make pre-push` which runs the linters